src/drivers/distance_sensor/ll40ls/LidarLite.h

```diff
@@ -46,8 +46,8 @@
 
 /* Device limits */
 #define LL40LS_MIN_DISTANCE (0.05f)
-#define LL40LS_MAX_DISTANCE_V3 (35.00f)
-#define LL40LS_MAX_DISTANCE_V1 (25.00f)
+#define LL40LS_MAX_DISTANCE (35.00f)
```

Comment:
I'm not sure about this change. The v1 sensors really don't work on grass at 35 meters and so you end up claiming that you have a good signal when you really will have a quite erratic reading. The argument provided in the PR itself is non-convincing, as the specs of LIDAR sensors tend to be measured against a material with 70% reflectance (which is a white sheet of paper). But real world materials are way, way below.

![Spectral-reflectance-of-different-materials-across-visible-to-infrared-wavelengths](https://user-images.githubusercontent.com/1208119/63400535-72527380-c3d4-11e9-83a3-423d00eea28d.png)

Reply:
The main issue in this driver is that there is not a clear way to identify different hardware versions.

In the actual upstream implementation (see [here](https://github.com/PX4/Firmware/blob/914417f5807344cbbdfd08745c769ecd413ba8b1/src/drivers/distance_sensor/ll40ls/LidarLiteI2C.cpp#L131-L156) ) v1 versions are distinguished based on the device address (**0x42** for **v1**, **0x62** otherwise), which I think is not properly correct.
As a matter of fact the [v1 datasheet](https://github.com/PulsedLight3D/LIDAR-Lite-Documentation/blob/master/Docs/LIDAR-Lite-v1-docs.pdf) reports that "_The sensor module has a 7­bit slave address with a default value of 0x62 in hexadecimal notation_", didn't find any reference to 0x42 address.
If we stick to this statement then the actual logic for distinguishing the v1 among the others is broken. I don't currently have any v1 hardware to be able to properly verify if this is the case.

The proposed implementation looks at hw_version, sw_version and device id which seemed to be more reliable. 
I did some tests with v2, v3, v3hp and the logic holds well, unfortunately I don't have data to catch the v1 case. 
This brought me to remove the V1 max distance as I think that this was never applied. 

Summing up I would go for one of the following solutions:
- If we verify that actually all the v1 sensors are on 0x42 address then I would re-introduce the address-based logic for setting the v1 limit.
- If the v1 have the usual 0x62 address then I would need to define a new case based on hw/sw version

Results of the probing:
v3:
INFO  [ll40ls] hw: 23, sw:1
INFO  [ll40ls] id: 0
INFO  [ll40ls] probe success - hw: 23, sw:1

v3HP:
INFO  [ll40ls] hw: 0, sw:200
INFO  [ll40ls] id: 13113
INFO  [ll40ls] probe success - id: 13113

v2:
INFO  [ll40ls] hw: 21, sw:2
INFO  [ll40ls] id: 23612
INFO  [ll40ls] probe success - hw: 21, sw:2

Reply:
Stupid question, and I appreciate this is a complex problem to solve: If we can identify v3, can't we just say those are 35 meters and the rest is 20 meters? Do we have proof that the v3 is indeed delivering 35 meters, or are we just hoping it improved?

Reply:
I actually did some further testing to verify if there is actually a common baseline between all the versions:
Flight was performed on grass in a cloudy day:

v2 (drop at 32m):
![v2](https://user-images.githubusercontent.com/7490213/63531814-995e9180-c509-11e9-8641-8e1276d6f625.png)

v3 (drop at 25m):
![v3](https://user-images.githubusercontent.com/7490213/63531825-a11e3600-c509-11e9-87f9-4a7527b9b395.png)

v3hp (drop at 16m):
![v3hp](https://user-images.githubusercontent.com/7490213/63531839-ada28e80-c509-11e9-8679-0c0779be4773.png)

For the v3 I have actually checked older logs and 25 meters are the best case; usually  they drop at about 15-20meters.

Seems that a good baseline would be 25 meters and 35 meters for the v2. What do you think?